### ISSUE TITLE
projects: `spec.resourceId` isn't respected when provided, `metadata.name` is used

### ISSUE THREAD

**Issue by aw185176** [GoogleCloudPlatform/k8s-config-connector#462]
## Checklist

- [x] I did not find a related open issue.
- [x] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [ ] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

## Bug Description
`spec.resourceId` isn't respected when provided, `metadata.name` is set.

## Additional Diagnostic Information

This is on version 1.49.1 deployed manually onto GKE

### Kubernetes Cluster Version
Server Version: v1.18.16-gke.2100

### Config Connector Version
1.49.1

### Config Connector Mode
cluster

### Steps to reproduce the issue

- create resource with spec.resourceId and metadata.name that are different
- schedule it, you'll get underlying resource not exists error etc 
- if you then move the same spec.resourceId to metadata.name, it'll create immediately, implying that the `spec.resourceId` was a valid identifier from the start

**Comment by jcanseco** [comment]
Hi @aw185176, thanks for bringing this issue up to us. I can confirm that I was able to reproduce the issue with the following error message:

```
Update call failed: underlying resource no longer exists and can't be recreated without creating 
a brand new resource
```

We should have a fix out shortly (within the next couple weeks).

**Comment by aw185176** [comment]
Awesome, thanks for the update.

**Comment by maqiuyujoyce** [comment]
The issue should be fixed in Config Connector [v1.50.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.50.0).

**Comment by KingBain** [comment]
Not sure this has been resolved. Having similar issue with latest version of CC 
https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/818
